buffer/src/main/java/io/netty/buffer/AbstractPooledDerivedByteBuf.java

```diff
@@ -124,4 +131,85 @@ public final ByteBuf retainedSlice() {
         final int index = readerIndex();
         return retainedSlice(index, writerIndex() - index);
     }
+
+    @Override
+    public ByteBuf slice(int index, int length) {
+        // All reference count methods should be inherited from this object (this is the "parent").
+        return new PooledNonRetainedSlicedByteBuf(unwrap(), index, length);
+    }
+
+    private final class PooledNonRetainedSlicedByteBuf extends UnpooledSlicedByteBuf {
+        PooledNonRetainedSlicedByteBuf(AbstractByteBuf buffer, int index, int length) {
+            super(buffer, index, length);
+        }
+
+        @Override
+        int refCnt0() {
+            return AbstractPooledDerivedByteBuf.this.refCnt();
+        }
+
+        @Override
+        ByteBuf retain0() {
+            AbstractPooledDerivedByteBuf.this.retain();
+            return this;
+        }
+
+        @Override
+        ByteBuf retain0(int increment) {
+            AbstractPooledDerivedByteBuf.this.retain(increment);
+            return this;
+        }
+
+        @Override
+        ByteBuf touch0() {
+            AbstractPooledDerivedByteBuf.this.touch();
+            return this;
+        }
+
+        @Override
+        ByteBuf touch0(Object hint) {
+            AbstractPooledDerivedByteBuf.this.touch(hint);
+            return this;
+        }
+
+        @Override
+        boolean release0() {
+            return AbstractPooledDerivedByteBuf.this.release();
+        }
+
+        @Override
+        boolean release0(int decrement) {
+            return AbstractPooledDerivedByteBuf.this.release(decrement);
+        }
+
+        @Override
+        public ByteBuf duplicate() {
+            // Capacity is not allowed to change for a sliced ByteBuf, so length == capacity()
+            final ByteBuf duplicate = slice(0, capacity());
+            duplicate.setIndex(readerIndex(), writerIndex());
+            return duplicate;
+        }
+
+        @Override
+        public ByteBuf retainedDuplicate() {
+            return PooledDuplicatedByteBuf.newInstance(unwrap(), this, readerIndex(), writerIndex());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Prefix this call to "unwrap" with "super.". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2388)